vulnfeeds/cpp/main.go

```diff
@@ -140,9 +194,72 @@ func MaybeGetSourceRepoFromDebian(mdir string, pkg string) string {
 	return ""
 }
 
+// Use the GitHub API to query the repository's language metadata to make the determination.
+func InScopeGitHubRepo(repoURL string) bool {
+	// TODO(apollock): Implement
+	return true
+}
+
+// Clone the repo and look for C/C++ files to make the determination.
+func InScopeGitRepo(repoURL string) bool {
+	// TODO(apollock): Implement
+	return true
+}
+
+// Looks at what the repo to determine if it contains code using an in-scope language
+func InScopeRepo(repoURL string) bool {
+	parsedURL, err := url.Parse(repoURL)
+	if err != nil {
+		Logger.Infof("Warning: %s failed to parse, skipping", repoURL)
+		return false
+	}
+
+	switch parsedURL.Hostname() {
+	case "github.com":
+		return InScopeGitHubRepo(repoURL)
+	default:
+		return InScopeGitRepo(repoURL)
+	}
+}
+
+// Takes an NVD CVE record and outputs an OSV file in the specified directory.
+func CVEToOSV(CVE cves.CVEItem, repo, directory string) {
+	CPEs := cves.CPEs(CVE)
+	CPE, ok := cves.ParseCPE(CPEs[0])
+	if !ok {
+		Logger.Fatalf("Can't generate an OSV record for %s without CPE data", CVE.CVE.CVEDataMeta.ID)
```

Comment:
Can address in a follow up PR: Generally we prefer returning errors and letting the caller handle this rather than erroring out in Go.

Reply:
Ack. Bear in mind that this is still proof-of-concept and from talking with @another-rex the plan of record is to move a lot of the functions from this into a library and then extend `vulnfeeds/cmd/combine-to-osv/main.go` to also do what's being done with this.